modules/serverless-pingsource-yaml.adoc

```diff
@@ -0,0 +1,92 @@
+// Module included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-pingsource.adoc
+
+The following sections describe how to create, verify and remove a basic PingSource using YAML files.
+
+.Prerequisites
+
+* You have Knative Serving and Eventing installed.
+
+.Procedure
+
+. To verify that `PingSource` is working, create a simple Knative
+service that dumps incoming messages to its log.
+
+. Copy the example YAML into a file named `service.yaml`:
++
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+        - image: quay.io/openshift-knative/knative-eventing-sources-event-display:v0.13.2
+----
++
+. Create the service by entering the command:
++
+----
+$ oc apply --filename service.yaml
+----
+
+. For each set of ping events that you want to request, create a PingSource in the same namespace as the event consumer.
++
+Copy the example YAML into a file named `ping-source.yaml`:
++
+----
+apiVersion: sources.knative.dev/v1alpha2
+kind: PingSource
+metadata:
+  name: test-ping-source
+spec:
+  schedule: "*/2 * * * *"
+  jsonData: '{"message": "Hello world!"}'
+  sink:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+----
++
+Create the PingSource by entering the command:
++
+----
+$ oc apply --filename ping-source.yaml
+----
+
+. Check that the controller is mapped correctly by entering the following command and inspecting the output:
++
+----
+$ oc get pingsource.sources.knative.dev test-ping-source -oyaml
+----
++
+The output should be similar to:
+
+// check command, add sample output
```

Comment:
Check this command is correct; need sample output

Reply:
You might want to get rid of status fields.
```
apiVersion: sources.knative.dev/v1alpha2
kind: PingSource
metadata:
  annotations:
    sources.knative.dev/creator: minikube-user
    sources.knative.dev/lastModifier: minikube-user
  creationTimestamp: "2020-04-07T16:11:14Z"
  generation: 1
  name: test-ping-source
  namespace: default
  resourceVersion: "55257"
  selfLink: /apis/sources.knative.dev/v1alpha2/namespaces/default/pingsources/test-ping-source
  uid: 3d80d50b-f8c7-4c1b-99f7-3ec00e0a8164
spec:
  jsonData: '{ value: "hello" }'
  schedule: '*/2 * * * *'
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: event-display
      namespace: default
status:
  conditions:
  - lastTransitionTime: "2020-04-07T16:11:21Z"
    status: "True"
    type: Deployed
  - lastTransitionTime: "2020-04-07T16:11:14Z"
    severity: Info
    status: "True"
    type: EventTypeProvided
  - lastTransitionTime: "2020-04-07T16:11:21Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2020-04-07T16:11:14Z"
    severity: Info
    status: "True"
    type: ResourcesCorrect
  - lastTransitionTime: "2020-04-07T16:11:14Z"
    status: "True"
    type: SinkProvided
  - lastTransitionTime: "2020-04-07T16:11:14Z"
    status: "True"
    type: ValidSchedule
  observedGeneration: 1
  sinkUri: http://event-display.default.svc.cluster.local
```

Reply:
You might want to get rid off status fields.
```
apiVersion: sources.knative.dev/v1alpha2
kind: PingSource
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"sources.knative.dev/v1alpha2","kind":"PingSource","metadata":{"annotations":{},"name":"test-ping-source","namespace":"default"},"spec":{"jsonData":"{\"message\": \"Hello world!\"}","schedule":"*/2 * * * *","sink":{"ref":{"apiVersion":"serving.knative.dev/v1","kind":"Service","name":"event-display"}}}}
    sources.knative.dev/creator: minikube-user
    sources.knative.dev/lastModifier: minikube-user
  creationTimestamp: "2020-04-07T16:21:21Z"
  generation: 1
  name: test-ping-source
  namespace: default
  resourceVersion: "56281"
  selfLink: /apis/sources.knative.dev/v1alpha2/namespaces/default/pingsources/test-ping-source
  uid: 2780ee43-0c06-4c21-8946-bf171afcc569
spec:
  jsonData: '{"message": "Hello world!"}'
  schedule: '*/2 * * * *'
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: event-display
status:
  conditions:
  - lastTransitionTime: "2020-04-07T16:21:23Z"
    status: "True"
    type: Deployed
  - lastTransitionTime: "2020-04-07T16:21:21Z"
    severity: Info
    status: "True"
    type: EventTypeProvided
  - lastTransitionTime: "2020-04-07T16:21:23Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2020-04-07T16:21:21Z"
    severity: Info
    status: "True"
    type: ResourcesCorrect
  - lastTransitionTime: "2020-04-07T16:21:21Z"
    status: "True"
    type: SinkProvided
  - lastTransitionTime: "2020-04-07T16:21:21Z"
    status: "True"
    type: ValidSchedule
  observedGeneration: 1
  sinkUri: http://event-display.default.svc.cluster.local

```